sbcq-17-copilot-instructions (#20)
bcgov/citz-sbc-queue: .github/copilot-instructions.md

```
src/
├── app/                # Next.js App Router pages & routes
├── components/         # Reusable UI components
├── hooks/              # App-specific React hooks
├── middleware/         # Next.js middleware functions
├── stores/             # Zustand store definitions
├── styles/             # Global styles & BC Gov design system
├── test/               # Test utilities and setup
└── utils/              # Utility functions
```

> `middleware/` contains Next.js middleware for authentication, routing, etc.
> `src/styles/` contains BC Government design system CSS files.

> > `src/hooks/` contains app-specific hooks like usePermissions.

Author: removed

  ```
- Use Image and Link from Next.js instead of native tags.

---

## BC Government Standards

### Design System & Branding

> - Follow **BC Government Design System** guidelines and standards
> - Use **predefined BC Government color classes** from `src/styles/bcgov/colours.css`:
>   - `bg-blue`, `text-blue` for primary blue (`#013366`)
>   - `bg-gold`, `text-gold` for accent gold (`#fcba19`)
>   - `bg-typography-primary`, `text-typography-primary` for text colors
>   - Avoid hardcoding hex values - use the Tailwind-integrated color classes
> - Implement **BC Government typography** using BC Sans font family
> - Maintain **consistent branding** with BC Government visual identity
> - Include **BC Government logos** and assets from `/public/bcgov/` directory

Reviewer: ```suggestion
Follow **BC Government Design System** guidelines and standards
- Use **predefined BC Government color classes** from `src/styles/bcgov/colours.css`
  - Avoid hardcoding hex values - use the Tailwind-integrated color classes
- Use **predefined BC Government button classes** from `src/styles/bcgov/components.css`
  - Most commonly used button classes are `primary` and `secondary`
```

No need to mention specific colors if we're giving it the file path. Too specific information and it might get confused. We also dont need to mention the BCSans font since its already applied across the application.

It also won't know what BC Government design standards are, it would have to search the web for that and copilot instructions wont do that.

### Font Integration
- Font weights available: Regular, Italic, Bold, BoldItalic
- Fonts served from `/public/fonts/` directory as WOFF2 files

> - Follow **BC Sans usage guidelines** for government applications

Author: removed

### Government Requirements
- Ensure **bilingual considerations** for future French language support
- Follow **accessibility standards** required for government services (WCAG 2.1 AA)
- Implement **government-standard navigation** patterns and information architecture
- Consider **privacy and security** requirements for government applications

---

## TypeScript Practices

- Use `type` (not `interface`) for consistency.
- Avoid `any`; use `unknown`, `Record<string, unknown>`, etc.
- Prefer `Partial`, `Pick`, `Omit`, and discriminated unions.

> - Always type props as `Props`, and all functions and stores explicitly.

Reviewer: ```suggestion
- Always type component props as `<component-name>Props`, such as `LoginButtonProps` for the `LoginButton` component.
```

- Small, focused components.
- Separate all logic from JSX.
- Include proper loading and error states.
- **Component props structure**:
  - Props should be an object with type `Props`
  - `Props` type should be defined and exported within the component file
  - Example:
    ```tsx
    export type Props = {
      title: string
      isVisible: boolean
      onClose: () => void
    }

    export default function Modal(props: Props) {
      const { title, isVisible, onClose } = props
      // Component implementation
    }
    ```
- **Accessibility-first development**:
  - Use semantic HTML elements by default
  - Include proper ARIA attributes for complex interactions
  - Ensure all interactive elements are keyboard accessible
  - Provide meaningful labels and descriptions
- Export:

>   - Components: default export

Reviewer: ```suggestion
  - Components: named and default export
```

- Small, focused components.
- Separate all logic from JSX.
- Include proper loading and error states.

> - **Component props structure**:
>   - Props should be an object with type `Props`
>   - `Props` type should be defined and exported within the component file
>   - Example:
>     ```tsx
>     export type Props = {
>       title: string
>       isVisible: boolean
>       onClose: () => void
>     }
> 
>     export default function Modal(props: Props) {
>       const { title, isVisible, onClose } = props
>       // Component implementation
>     }
>     ```

Reviewer: ```suggestion
- **Component props structure**:
  - Props type should be defined and exported within the component file
  - Props should be destructured within the component parameter list.
```

We're repeating the part about the type of props. Instead lets use this section to define how its used and exported and leave the typescript part to the above section.

## Styling & Accessibility

- Use **TailwindCSS v4** as the primary styling framework with utility-first approach

> - **BC Government design system** colors are predefined in `src/styles/bcgov/colours.css`:
>   - Use BC Gov color classes: `bg-blue`, `text-gold`, `bg-typography-primary`, etc.
>   - Colors defined using `@theme` directive for Tailwind integration
>   - Spacing standards available in `src/styles/bcgov/spacing.css`

Reviewer: ```suggestion
- Use BC Government design guidelines as listed above.
```

We were repeating what was said earlier about bc gov styling.
